e2e/ui/Enterprise SSO/setup_link.spec.ts

```diff
@@ -21,96 +21,171 @@ export const test = baseTest.extend<MyFixtures>({
-    await page.getByLabel('Edit').click();
-    await page.getByRole('button', { name: 'Delete' }).click();
-    await page.getByRole('button', { name: 'Confirm' }).click();
+    // use setup link in all tests
+    await use(setuplinkPage);
 
     // remove setup link
     await setuplinkPage.removeSetupLink();
+  },
+});
+
+const setupLinkTestsData: {
+  testDescription: string;
+  testTitle: string;
+  testRunner: (setupLinkPage: Page) => Promise<void>;
+}[] = [
+  {
+    testDescription: 'Admin Portal Enterprise SSO SetupLink using generic SAML 2.0',
+    testTitle: 'should be able to create setup link and sso connection using generic SAML 2.0',
+    testRunner: testGenericSaml,
+  },
+  {
+    testDescription: 'AAdmin Portal Enterprise SSO SetupLink using PingOne SAML 2.0',
+    testTitle: 'should be able to create setup link and sso connection using PingOne SAML 2.0',
+    testRunner: testPingOneSaml,
+  },
+];
+
+setupLinkTestsData.forEach((testData) => {
+  test.describe(testData.testDescription, () => {
+    test(testData.testTitle, async ({ page, setuplinkPage }) => {
+      // get setuplink url
+      const linkContent = await setuplinkPage.getSetupLinkUrl();
+
+      // Open new tab and go to setup link page
+      const context = page.context();
+      const setupLinkPage = await context.newPage();
+      await setupLinkPage.goto(linkContent);
+
+      // run generic saml 2.0 test
+      await testData.testRunner(setupLinkPage);
+
+      await setupLinkPage.waitForURL(/\/setup\/.+\/sso-connection$/);
+      await expect(setupLinkPage.getByRole('cell', { name: 'saml.example.com' })).toBeVisible();
+      await setupLinkPage.close();
+
+      // Go to connections page
+      await page.goto(TEST_SETUPLINK_ADMIN_CONNECTION);
+
+      // Check if new SSO connection is created
+      await expect(
+        page.getByText(TENANT, { exact: true }),
+        'Failed to create new sso connection from setup-link'
+      ).toBeVisible();
+      await expect(
+        page.getByText(PRODUCT, { exact: true }),
+        'Failed to create new sso connection from setup-link'
+      ).toBeVisible();
+
+      // Delete the SSO connection
+      await page.getByLabel('Edit').click();
+      await page.getByRole('button', { name: 'Delete' }).click();
+      await page.getByRole('button', { name: 'Confirm' }).click();
+    });
   });
 });
+
+async function testGenericSaml(setupLinkPage: Page) {
+  // Create SSO connection using generic SAML 2.0 workflow
+  await setupLinkPage.getByRole('button', { name: 'Generic SAML 2.0' }).click();
+
+  // check mdx generated content using remart-gfm plugin for step1
+  await expect(setupLinkPage.getByRole('heading', { name: 'Step 1: Configuration SAML' })).toBeVisible();
+  let p1 = await setupLinkPage.getByText('Your Identity Provider (IdP)').textContent();
+  expect(
+    p1 ===
+      'Your Identity Provider (IdP) will ask for the following information while configuring the SAML application.'
+  ).toBeTruthy();
```

Comment:
The same assertion can be done using [toHaveText](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-text) in a single line.